### PR TITLE
added Maxwell unit

### LINKED ISSUE #12969: Adding a new CGS unit: Maxwell
### Description

Recently, I was trying to parse a unit that's quite commonly used in solar physics: `Mx` (a Maxwell) which is a magnetic flux in CGS.

I was unable to find any discussions or issues about this from a quick search, so would anyone object to me adding it in a PR?

### PR DESCRIPTION
**Pull request by nabobalis**
### Description

This pull request is to address adding a missing CGS unit, Maxwell.

I am unsure what tests to add or where. So those are missing but I am more than happy to add as many as needed.

I also added a changelog file, I hope its ok.

Fixes #12969

### Checklist for package maintainer(s)

This checklist is meant to remind the package maintainer(s) who will review this pull request of some common things to look for. This list is not exhaustive.

- [x] Do the proposed changes actually accomplish desired goals?
- [x] Do the proposed changes follow the [Astropy coding guidelines](https://docs.astropy.org/en/latest/development/codeguide.html)?
- [x] Are tests added/updated as required? If so, do they follow the [Astropy testing guidelines](https://docs.astropy.org/en/latest/development/testguide.html)?
- [x] Are docs added/updated as required? If so, do they follow the [Astropy documentation guidelines](https://docs.astropy.org/en/latest/development/docguide.html#astropy-documentation-rules-and-guidelines)?
- [x] Is rebase and/or squash necessary? If so, please provide the author with appropriate instructions. Also see ["When to rebase and squash commits"](https://docs.astropy.org/en/latest/development/when_to_rebase.html).
- [x] Did the CI pass? If no, are the failures related? If you need to run daily and weekly cron jobs as part of the PR, please apply the `Extra CI` label.
- [x] Is a change log needed? If yes, did the change log check pass? If no, add the `no-changelog-entry-needed` label. If this is a manual backport, use the `skip-changelog-checks` label unless special changelog handling is necessary.
- [x] Is this a big PR that makes a "What's new?" entry worthwhile and if so, is (1) a "what's new" entry included in this PR and (2) the "whatsnew-needed" label applied?
- [x] Is a milestone set? Milestone must be set but `astropy-bot` check might be missing; do not let the green checkmark fool you.
- [x] At the time of adding the milestone, if the milestone set requires a backport to release branch(es), apply the appropriate `backport-X.Y.x` label(s) *before* merge.

### CODE REVIEW

**Comment by pllim** [comment]
Thanks! I'll let the subpackage maintainers decide if this needs a proper unit test or not.

File: astropy/units/cgs.py
Comment (mhvk) [inline astropy/units/cgs.py:111]: Do we want to have prefixes? I.e., is it common use to have `mMx` or `kMx`?

**Review by mhvk — COMMENTED** [review]
Thanks!  This looks good modulo my question about whether we want to allow prefixes. Am happy either way, but just good to check.

File: astropy/units/cgs.py
Comment (nabobalis) [reply astropy/units/cgs.py:111]: At the time I was like yeah but checking some papers that use them, they keep it at Mx and they do 2x10^6 Mx instead.

Comment (nabobalis) [inline astropy/units/cgs.py:111]:
```suggestion
def_unit(['Mx', 'Maxwell', 'maxwell'], 1e-8 * si.Wb, namespace=_ns,
```

**Review by mhvk — APPROVED** [review]
OK, this looks good to me now! Thanks again! 

I'll approve but before merging, @pllim, the failures all seem unrelated, but the readthedocs one is weird - is this a currently known problem?

**Comment by pllim** [comment]
Hmm try rebase.

**Review by nstarman — COMMENTED** [review]
LGTM! Thanks. I'll check in again when RTD is building. Just want to confirm that the new unit is correctly showing up in the tables of units.

**Comment by nabobalis** [comment]
Rebased.

**Comment by pllim** [comment]
IERS failure is unrelated.

**Review by nstarman — APPROVED** [review]
Screen Shot 2022-03-28 at 16 37 43

**Comment by nabobalis** [comment]
Thank you all for your comments and reviews!

**Comment by pllim** [comment]
Thanks!

**Pull request merged**
